### PR TITLE
remove some unused storage engine APIs

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Remove now-unused storage engine APIs

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *all tests*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9316/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests ok except scache-related build errors.

File: arangod/VocBase/LogicalView.cpp [hunk 1]
```diff
@@ -405,27 +405,7 @@ Result LogicalView::rename(std::string&& newName) {
     return Result();  // NOOP
```
Comment (ObiWahn) [inline arangod/VocBase/LogicalView.cpp:405]: Is this return still needed?

Comment (jsteemann) [reply arangod/VocBase/LogicalView.cpp:405]: Good catch. I think it can be safely removed as well. Will adjust the PR.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9329/

**Pull request merged**